### ISSUE TITLE
noImplicitAny should warn/error if an array declaration [] has no type annotation

### ISSUE THREAD

**Issue by safareli** [microsoft/TypeScript#43752]
# Bug Report

### 🔎 Search Terms

any[] unknown[] array declaration strict infer inferred  no implicit any noimplicitany

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about any, unknown, strict, implicit

### 💻 Code

If you have `const arr = []` 

### 🙁 Actual behavior

Inferred type will be `arr: any[]`, even if you turn on existing strict flags  you would get 0 warnings/errors. 

### 🙂 Expected behavior

if you have all strict flags, on especially `noImplicitAny` you would expect inferred type to be `unknown[]` or at least a _warning_.

**Comment by nmain** [comment]
Arrays declared this way have evolving types, and I don't think it lets you do anything unnatural with the array without warning.  Pushes result in an evolving `(T1 | T2 | ...)[]` type, and if you tried to send that to something else which only accepted a `T1[]` you'd get a failure at that other assignment, which makes it consistent with other unannotated declarations.

**Comment by safareli** [comment]
Thanks for clarification.

This compiles:

```ts
const kitri = []
kitri.push(1)
const x = kitri[0]
kitri.push(x)
```

this doesn't
```ts
const kitri = []
// kitri.push(1)
const x = kitri[0]
kitri.push(x)
```
👍
